src/util/performance/PerformanceUtil.cpp

```diff
@@ -228,3 +349,404 @@ int PerformanceUtil::collectLocalPerformanceData(std::string isVMStatManager, st
+                write(sockfd, isResourceAllocationRequired.c_str(), isResourceAllocationRequired.size());
+                scheduler_logger.log("Sent : Resource Allocation Requested " + isResourceAllocationRequired, "info");
+
+                bzero(data, 301);
+                read(sockfd, data, 300);
+                response = (data);
+                response = utils.trim_copy(response, " \f\n\r\t\v");
+
+                scheduler_logger.log("Performance Response " + response, "info");
+
+                std::vector<std::string> strArr = Utils::split(response, ',');
+
+                std::string processTime = strArr[0];
+                std::string memoryUsage = strArr[1];
+                std::string cpuUsage = strArr[2];
+
+
+                placeResourceConsumption.memoryUsage = atoi(memoryUsage.c_str());
+                placeResourceConsumption.host = host;
+
+                return placeResourceConsumption;
+            }
+        }
+    }
+}
+
+bool PerformanceUtil::isResourcesSufficient(std::string graphId, std::string command, std::string category) {
+    PerformanceUtil performanceUtil;
+    performanceUtil.init();
+    bool resourcesSufficient = true;
+    std::set<std::string> hostSet;
+    std::vector<ResourceConsumption> placeResouceConsumptionList = performanceUtil.retrieveCurrentResourceUtilization();
+
+    string sqlStatement = "SELECT worker_idworker, name,ip,user,server_port,server_data_port,partition_idpartition "
+                          "FROM worker_has_partition INNER JOIN worker ON worker_has_partition.worker_idworker=worker.idworker "
+                          "WHERE partition_graph_idgraph=" + graphId + ";";
+
+    std::vector<vector<pair<string, string>>> results = sqlLiteDB.runSelect(sqlStatement);
+
+    int partitionCount = results.size();
+
+    for (std::vector<vector<pair<string, string>>>::iterator i = results.begin(); i != results.end(); ++i) {
+        std::vector<pair<string, string>> rowData = *i;
+        string ip = rowData.at(2).second;
+
+        hostSet.insert(ip);
+    }
+
+    for (std::set<std::string>::iterator hostSetIterator = hostSet.begin(); hostSetIterator != hostSet.end(); ++hostSetIterator) {
+        std::string host = *hostSetIterator;
+        std::vector<ResourceConsumption>::iterator consumptionIterator;
+
+        for (consumptionIterator = placeResouceConsumptionList.begin(); consumptionIterator != placeResouceConsumptionList.end(); ++consumptionIterator) {
+            ResourceConsumption consumption = *consumptionIterator;
+
+            if (host == consumption.host) {
+                std::string hostMemoryAllocation;
+                int currentMemoryUsage = consumption.memoryUsage;
+
+                std::string hostMemoryAllocationQuery = "select total_memory from host where ip='" + host + "';";
+                std::string slaMemoryRequirementQuery = "select memory_usage from graph_place_sla_performance INNER JOIN "
```

Comment:
There is an issue with the line indentation in this query string. Lets fix that.

Reply:
Code formatting was corrected.